File: src/AdventureWorks.Application/DataEngine/Purchasing/ProductVendor/Commands/Queries/GetProductVendors/ProductVendorLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.Purchasing;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;
using AdventureWorks.Application.DataEngine.Production.Product.Queries.GetProducts;
using AdventureWorks.Application.DataEngine.Production.UnitMeasure.Queries.GetUnitMeasures;
namespace AdventureWorks.Application.DataEngine.Purchasing.ProductVendor.Queries.GetProductVendors
{
    public partial class ProductVendorLookupModel : IHaveCustomMapping
    {
        public int ProductID { get; set; }
        public int VendorID { get; set; }
        public int AverageLeadTime { get; set; }
        public decimal StandardPrice { get; set; }
        public decimal? LastReceiptCost { get; set; }
        public DateTime? LastReceiptDate { get; set; }
        public int MinOrderQty { get; set; }
        public int MaxOrderQty { get; set; }
        public int? OnOrderQty { get; set; }
        public string UnitMeasureCode { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.Purchasing.ProductVendor, ProductVendorLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<ProductVendorLookupModel, BaseProductVendor>().IgnoreMissingDestinationMembers();
        }
    }
}

File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/purchasing/product-vendor/product-vendor-lookup-model.ts

export interface IProductVendorLookupModel {
  productID: number;
  vendorID: number;
  averageLeadTime: number;
  standardPrice: number;
  lastReceiptCost: number | null;
  lastReceiptDate: Date | string | null;
  minOrderQty: number;
  maxOrderQty: number;
  onOrderQty: number | null;
  unitMeasureCode: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class ProductVendorLookupModel implements IProductVendorLookupModel {
  productID: number;
  vendorID: number;
  averageLeadTime: number;
  standardPrice: number;
  lastReceiptCost: number | null;
  lastReceiptDate: Date | string | null;
  minOrderQty: number;
  maxOrderQty: number;
  onOrderQty: number | null;
  unitMeasureCode: string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<ProductVendorLookupModel>) {
    if (!!init) {
      Object.assign<ProductVendorLookupModel, Partial<ProductVendorLookupModel>>(this, init);
    }
  }
}
